### PR TITLE
[Shadowlands][Warlock] Stat Boxes and Integration Tests

### PR DESCRIPTION
**Pull request by Sharrq**
Updated the Warlock statistic boxes to the new format and added an integration test for all 3 specs. Also removed the BFA Azerite Trait modules.

Towards #3739

### CODE REVIEW

**Review by MartijnHols — APPROVED** [review]
Sweet, thanks!

File: src/parser/warlock/affliction/modules/soulshards/SoulShardDetails.js [hunk 1]
```diff
@@ -48,14 +48,12 @@ class SoulShardDetails extends Analyzer {
     return (
       <Statistic
         position={STATISTIC_ORDER.CORE(3)}
-        size="small"
+        size="flexible"
+        tooltip={(<>In order for Focus Magic to compete with the other talents on that row, you need to ensure you are getting as much uptime out of the buff as possible. Therefore, if you forget to put the buff on another player or if they player you gave it to is not getting crits very often, then you might need to consider giving the buff to someone else. Ideally, you should aim to trade buffs with another mage who has also taken Focus Magic so you both get the full benefit.</>)}
```
Comment (MartijnHols) [inline src/parser/warlock/affliction/modules/soulshards/SoulShardDetails.js:52]: Oof

File: src/parser/warlock/demonology/modules/soulshards/SoulShardDetails.js [hunk 2]
```diff
@@ -44,16 +44,15 @@ class SoulShardDetails extends Analyzer {
...
-            alt="Wasted Soul Shards"
-          />
-        )}
-        value={shardsWasted}
-        label="Wasted Soul Shards"
-      />
+      <Statistic
+        position={STATISTIC_ORDER.CORE(3)}
+        size="flexible"
+        tooltip={(<>In order for Focus Magic to compete with the other talents on that row, you need to ensure you are getting as much uptime out of the buff as possible. Therefore, if you forget to put the buff on another player or if they player you gave it to is not getting crits very often, then you might need to consider giving the buff to someone else. Ideally, you should aim to trade buffs with another mage who has also taken Focus Magic so you both get the full benefit.</>)}
```
Comment (MartijnHols) [inline src/parser/warlock/demonology/modules/soulshards/SoulShardDetails.js:50]: oof2

File: src/parser/warlock/destruction/modules/soulshards/SoulShardDetails.js [hunk 3]
```diff
@@ -47,18 +47,16 @@ class SoulShardDetails extends Analyzer {
...
 
   statistic() {
-    const fragmentsWasted = this.soulShardTracker.wasted;
+    const shardsWasted = this.soulShardTracker.wasted;
     return (
       <Statistic
         position={STATISTIC_ORDER.CORE(3)}
-        size="small"
+        size="flexible"
+        tooltip={(<>In order for Focus Magic to compete with the other talents on that row, you need to ensure you are getting as much uptime out of the buff as possible. Therefore, if you forget to put the buff on another player or if they player you gave it to is not getting crits very often, then you might need to consider giving the buff to someone else. Ideally, you should aim to trade buffs with another mage who has also taken Focus Magic so you both get the full benefit.</>)}
```
Comment (MartijnHols) [inline src/parser/warlock/destruction/modules/soulshards/SoulShardDetails.js:55]: oof3

File: src/parser/warlock/affliction/modules/soulshards/SoulShardDetails.js (hunk 1 above)
Comment (Sharrq) [reply src/parser/warlock/affliction/modules/soulshards/SoulShardDetails.js:52]: oops ..... Ill fix it and throw in another PR

**Pull request merged**